### PR TITLE
Add kalman smoother and log likelihood

### PR DESCRIPTION
**Pull request by Shunsuke-Hori**
This PR adds Kalman smoother and a function to compute log-likelihood. I also made minor updates to existing code.

Ready for review

PS: I'm wondering if the current `kalman.jl` is efficient and well documented enough.

### CODE REVIEW

File: src/kalman.jl [hunk 1]
```diff
@@ -124,3 +124,109 @@ function stationary_values(k::Kalman)
...
+
+"""
+computes log-likelihood of period ``t``
+
+##### Arguments
+- `kn::Kalman`: `Kalman` specifying the model. Current values must be the 
+                forecast for period ``t`` observation conditional on ``t-1``
+                observation.
+- `y::AbstractVector`: observations at period ``t``
+##### Returns
```
Comment (cc7768) [inline src/kalman.jl]: It improves readability to put a new line to separate the end of the inputs and the beginning of the returns.

There are a few other places this shows up in this file.

File: src/kalman.jl [hunk 2]
```diff
@@ -71,8 +71,8 @@ function prior_to_filtered!(k::Kalman, y)
         reshape(y, k.k, 1)
     end
     A = Sigma * G'
-    B = G * Sigma' * G' + R
-    M = A * inv(B)
+    B = G * Sigma * G' + R
+    M = A / B
```
Comment (cc7768) [inline src/kalman.jl]: Good catch! This is better than taking the inverse.

**Comment by Shunsuke-Hori** [comment]
@cc7768 updated. Thank you!!

**Comment by Flintro** [comment]
I double checked the math and equations. Also looks good to me. Really cool contribution all round as well.

**Pull request merged**